src/api/historyApi.ts

```diff
@@ -0,0 +1,37 @@
+import {TransactionType} from '../models';
+import {RestApi} from './index';
+import {ApiResponse} from './types/index';
+
+export interface HistoryApi {
+  fetchWalletHistory: (
+    walletId: string,
+    skip: number,
+    take: number,
+    assetId?: string,
+    operationType?: TransactionType[]
+  ) => ApiResponse<any>;
+}
+
+export class RestHistoryApi extends RestApi implements HistoryApi {
+  fetchWalletHistory = (
+    walletId: string,
+    skip: number,
+    take: number,
+    assetId?: string,
+    operationType?: TransactionType[]
+  ) => {
+    const query: any = {skip, take};
+
+    if (assetId) {
```

Comment:
i'm not sure you need this additional checks. it may be `getWithQuery` omits undefined params